components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/inbound/InboundFunctions.java

```diff
@@ -0,0 +1,165 @@
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Utility functions related to application inbound protocols.
+ */
+public class InboundFunctions {
+    
+    private static final Log log = LogFactory.getLog(InboundFunctions.class);
+    
+    private InboundFunctions() {
+    
+    }
+    
+    /**
+     * Get the inbound authentication key for the given inbound type.
+     *
+     * @param application Service provider.
+     * @param inboundType Inbound type.
+     * @return Inbound authentication key.
+     */
+    public static String getInboundAuthKey(ServiceProvider application,
+                                           String inboundType) {
+        
+        InboundAuthenticationConfig inboundAuthConfig = application.getInboundAuthenticationConfig();
+        if (inboundAuthConfig != null) {
+            InboundAuthenticationRequestConfig[] inbounds = inboundAuthConfig.getInboundAuthenticationRequestConfigs();
+            if (inbounds != null) {
+                return Arrays.stream(inbounds)
+                        .filter(inbound -> inboundType.equals(inbound.getInboundAuthType()))
+                        .findAny()
+                        .map(InboundAuthenticationRequestConfig::getInboundAuthKey)
+                        .orElse(null);
+            }
+        }
+        return null;
+    }
+    
+    /**
+     * Rollback the inbounds.
+     *
+     * @param currentlyAddedInbounds Currently added inbounds list.
+     * @throws IdentityApplicationManagementException If an error occurred while rolling back the inbounds.
+     */
+    public static void rollbackInbounds(List<InboundAuthenticationRequestConfig> currentlyAddedInbounds)
+            throws IdentityApplicationManagementException {
+        
+        for (InboundAuthenticationRequestConfig inbound : currentlyAddedInbounds) {
+            rollbackInbound(inbound);
+        }
+    }
+    
+    /**
+     * Rollback the inbound authentication request config.
+     *
+     * @param inbound Inbound authentication request config.
+     * @throws IdentityApplicationManagementException If an error occurred while rolling back the inbound.
+     */
+    public static void rollbackInbound(InboundAuthenticationRequestConfig inbound)
+            throws IdentityApplicationManagementException {
+        
+        List<ApplicationInboundAuthConfigHandler> applicationInboundAuthConfigHandlerList =
+                ApplicationManagementServiceComponentHolder.getInstance().getApplicationInboundAuthConfigHandler();
+        for (ApplicationInboundAuthConfigHandler applicationInboundAuthConfigHandler :
+                applicationInboundAuthConfigHandlerList) {
+            if (applicationInboundAuthConfigHandler.canHandle(inbound.getInboundAuthType())) {
+                applicationInboundAuthConfigHandler.handleConfigDeletion(inbound.getInboundAuthKey());
```

Comment:
Can we add a break here?

Reply:
Fixed with 51b85af647109d22ee800443e12e61cc0b09c1dc